### PR TITLE
Documented  the `cli` module

### PR DESCRIPTION
**Pull request by tarekziade**
## Part of https://github.com/elastic/enterprise-search-team/issues/4017

Adds doc for `cli`

### CODE REVIEW

File: connectors/cli.py [hunk 1]
```diff
@@ -86,6 +97,12 @@ def _parser():
 
 
 async def _start_service(config, loop):
+    """Starts the service.
+
+    Steps:
+    - performs a preflight checks using `PreflightCheck`
+    - instanciates a `MultiService` instance and run its `run` async function
```
Comment (artem-shelkovnikov) [inline connectors/cli.py:104]:
```suggestion
    - instantiates a `MultiService` instance and runs its `run` async function
```

File: connectors/cli.py [hunk 2]
```diff
@@ -86,6 +97,12 @@ def _parser():
 
 
 async def _start_service(config, loop):
+    """Starts the service.
+
+    Steps:
+    - performs a preflight checks using `PreflightCheck`
```
Comment (navarone-feekery) [inline connectors/cli.py:103]:
```suggestion
    - performs a preflight check using `PreflightCheck`
```

File: connectors/cli.py (hunk 1 above)
Comment (artem-shelkovnikov) [reply connectors/cli.py:104]: I'm not sure about the description here:
- should it explicitly explain what services it runs?
- should it mention the signal handling?

File: connectors/cli.py [hunk 3]
```diff
@@ -6,7 +6,15 @@
...
-Parses arguments and call run() with them.
+This is the main entry point of the framework. When the project is installed as
+a Python package, an `elastic-ingest` executable is added in the PATH.
+
+
+`main` is doing the following:
+
+- parses the arguments
+- reads the config file
+- starts the event loop against a series of services
```
Comment (artem-shelkovnikov) [inline connectors/cli.py:17]: Is this needed since the `main` function is already documented?

File: connectors/cli.py (hunk 2 above)
Comment (navarone-feekery) [reply connectors/cli.py:103]: I'm assuming it's one preflight check. If there are multiple then of course `performs preflight checks` would be better.

File: connectors/cli.py [hunk 4]
```diff
@@ -152,6 +173,11 @@ def run(args):
 
 
 def main(args=None):
+    """Main entry point.
+
+    Parses the arguments and calls `run` with them.
+    If `--version` is used, it displays the version and exits.
+    """
```
Comment (artem-shelkovnikov) [inline connectors/cli.py:180]: should this description focus more on usage of `main` function? 

Something along the lines with

"This function is the entry point to the Connectors Service, it's responsible for all operations that the service can do:

- Listing all connectors (when args.action == 'list')
- Running connector service with provided args and configuration file stored in config.yaml
"

File: connectors/cli.py (hunk 1 above)
Comment (wangch079) [reply connectors/cli.py:104]:
```suggestion
    - starts a series of services via a `MultiService` instance
```

File: connectors/cli.py (hunk 2 above)
Comment (tarekziade) [reply connectors/cli.py:103]: we are doing several steps but I would assume that `preflight check` is fine -- devs can go and look at `PreflightCheck` I assume

File: connectors/cli.py (hunk 3 above)
Comment (tarekziade) [reply connectors/cli.py:17]: The intent of the module docstring is to offer a short explanation of what this module does without having to look around inside the module. Ideally, it should not be duplicating too much content -- so yeah we could reduce this for sure.

File: connectors/cli.py (hunk 4 above)
Comment (tarekziade) [reply connectors/cli.py:180]: sounds good!

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**